watertap/unit_models/electrocoagulation.py

```diff
@@ -0,0 +1,991 @@
+                )
+                res = opt.solve(self, tee=slc.tee)
+
+        init_log.info("Initialization Step 2 {}.".format(idaeslog.condition(res)))
+        # ---------------------------------------------------------------------
+        # Release Inlet state
+        self.properties_in.release_state(flags, outlvl=outlvl)
+        init_log.info("Initialization Complete: {}".format(idaeslog.condition(res)))
+
+        if not check_optimal_termination(res):
+            raise InitializationError(f"Unit model {self.name} failed to initialize")
+
+    def calculate_scaling_factors(self):
+        super().calculate_scaling_factors()
+
+        if iscale.get_scaling_factor(self.electrode_thickness) is None:
+            iscale.set_scaling_factor(self.electrode_thickness, 1e3)
+
+        if iscale.get_scaling_factor(self.electrode_mass) is None:
+            iscale.set_scaling_factor(self.electrode_mass, 1)
+
+        if iscale.get_scaling_factor(self.electrode_gap) is None:
+            iscale.set_scaling_factor(self.electrode_gap, 10)
+
+        if iscale.get_scaling_factor(self.electrolysis_time) is None:
+            iscale.set_scaling_factor(self.electrolysis_time, 0.1)
+
+        if iscale.get_scaling_factor(self.applied_current) is None:
+            iscale.set_scaling_factor(self.applied_current, 1)
+
+        if iscale.get_scaling_factor(self.current_efficiency) is None:
+            iscale.set_scaling_factor(self.current_efficiency, 1)
+
+        if iscale.get_scaling_factor(self.cell_volume) is None:
+            iscale.set_scaling_factor(self.cell_volume, 0.1)
+
+        if iscale.get_scaling_factor(self.ohmic_resistance) is None:
+            iscale.set_scaling_factor(self.ohmic_resistance, 1e3)
+
+        if iscale.get_scaling_factor(self.charge_loading_rate) is None:
+            iscale.set_scaling_factor(self.charge_loading_rate, 1e-2)
+
+        if iscale.get_scaling_factor(self.current_density) is None:
+            iscale.set_scaling_factor(self.current_density, 1e-2)
+
+        if iscale.get_scaling_factor(self.cell_voltage) is None:
+            iscale.set_scaling_factor(self.cell_voltage, 0.1)
+
+        if iscale.get_scaling_factor(self.overpotential) is None:
+            iscale.set_scaling_factor(self.overpotential, 0.1)
+
+        sf = iscale.get_scaling_factor(self.charge_loading_rate)
+        iscale.constraint_scaling_transform(self.eq_charge_loading_rate, sf)
+
+        sf = iscale.get_scaling_factor(self.ohmic_resistance)
+        iscale.constraint_scaling_transform(self.eq_ohmic_resistance, sf)
+
+    def _get_stream_table_contents(self, time_point=0):
+        return create_stream_table_dataframe(
+            {
+                "Feed Inlet": self.inlet,
+                "Liquid Outlet": self.outlet,
+                "Waste Outlet": self.waste,
+            },
+            time_point=time_point,
+        )
+
+    def _get_performance_contents(self, time_point=0):
+
+        # TODO
+        var_dict = {}
```

Comment:
I see this is a todo. Would be good to populate some minimal set of bare essentials.